Stop exposing getMatchedCSSRules to the Web

cdumez opened the pull request:
#### 7d88064a3d1208788a7e02387424ac356b83d288
<pre>
Stop exposing getMatchedCSSRules to the Web
<a href="https://bugs.webkit.org/show_bug.cgi?id=79653">https://bugs.webkit.org/show_bug.cgi?id=79653</a>
&lt;rdar://16354443&gt;

Reviewed by Sam Weinig and Darin Adler.

Stop exposing getMatchedCSSRules to the Web. We still expose it to user worlds because
Safari is still using it for now.

This was never standardized and both Blink and Gecko do not support it.

* LayoutTests/fast/css/getMatchedCSSProperties-rule-mutation-expected.txt: Removed.
* LayoutTests/fast/css/getMatchedCSSProperties-rule-mutation.html: Removed.
* LayoutTests/fast/css/getMatchedCSSRules-crash-expected.txt: Removed.
* LayoutTests/fast/css/getMatchedCSSRules-crash.html: Removed.
* LayoutTests/fast/css/modify-ua-rules-from-javascript-expected.txt: Removed.
* LayoutTests/fast/css/modify-ua-rules-from-javascript.html: Removed.
* LayoutTests/fast/dom/StyleSheet/detached-parent-rule-without-wrapper-expected.txt:
* LayoutTests/fast/dom/StyleSheet/detached-parent-rule-without-wrapper.html:
* LayoutTests/fast/dom/Window/get-set-properties-expected.txt:
* LayoutTests/fast/dom/Window/get-set-properties.html:
* LayoutTests/fast/dom/Window/getMatchedCSSRules-null-crash-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-null-crash.html: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-invalid-pseudo-elements-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-invalid-pseudo-elements.html: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements-complex-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements-complex.html: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements.html: Removed.
* LayoutTests/fast/dom/Window/resources/WindowProperties.js:
* LayoutTests/fast/dom/Window/resources/window-properties.js:
* LayoutTests/fast/dom/Window/window-function-frame-getter-precedence-expected.txt:
* LayoutTests/fast/dom/Window/window-function-name-getter-precedence-expected.txt:
* LayoutTests/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/fast/dom/Window/window-property-shadowing-expected.txt:
* LayoutTests/fast/dom/Window/window-property-shadowing.html:
* LayoutTests/fast/dom/collection-iterators.html:
* LayoutTests/fast/dom/defaultView-expected.txt:
* LayoutTests/fast/dom/defaultView.html:
* LayoutTests/fast/dom/resources/wrapper-identity-base.js:
* LayoutTests/fast/dom/wrapper-identity-expected.txt:
* LayoutTests/fast/inspector-support/matchedrules.html: Removed.
* LayoutTests/fast/selectors/ignore-pseudo-element-inside-non-rightmost-fragments-expected.txt: Removed.
* LayoutTests/fast/selectors/ignore-pseudo-element-inside-non-rightmost-fragments.html: Removed.
* LayoutTests/http/tests/security/cross-frame-access-call-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-call.html:
* LayoutTests/http/tests/security/cross-frame-access-get-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-get.html:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor.html:
* LayoutTests/http/tests/security/cross-frame-access-put-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-put.html:
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules.html: Removed.
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules2-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules2.html: Removed.
* LayoutTests/http/tests/security/resources/cross-frame-iframe-for-put-test.html:
* LayoutTests/http/tests/security/resources/cross-origin-getMatchedCSSRules-frame.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/compat/historical-expected.txt:
* LayoutTests/js/dom/global-function-resolve-expected.txt:
* LayoutTests/js/dom/script-tests/global-function-resolve.js:
* LayoutTests/platform/gtk/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/platform/mac-wk2/fast/dom/Window/window-lookup-precedence-expected.txt:
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
(WebCore::DOMWrapperWorld::isUser const):
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/page/DOMWindow.idl:

Canonical link: <a href="https://commits.webkit.org/252542@main">https://commits.webkit.org/252542@main</a>
</pre>

darinadler commented on Source/WebCore/page/DOMWindow.idl:
```diff
@@ -114,9 +114,8 @@
-    // Non-standard: should probably deprecate this (https://bugs.webkit.org/show_bug.cgi?id=79653).
-    // Blink already deprecated it (https://bugs.chromium.org/p/chromium/issues/detail?id=437569).
-    CSSRuleList getMatchedCSSRules(optional Element? element = null, optional DOMString? pseudoElement = null);
+    // Non-standard: We only expose this to user worlds because Safari is using it.
```
I think the comment should be more like:
`// Non-standard: For now we expose this to user worlds since Safari is using it, but we plan to remove it.`

cdumez replied:
Fixed.

cdumez committed:
Stop exposing getMatchedCSSRules to the Web
https://bugs.webkit.org/show_bug.cgi?id=79653
<rdar://16354443>

Reviewed by Sam Weinig and Darin Adler.

Stop exposing getMatchedCSSRules to the Web. We still expose it to user worlds because
Safari is still using it for now.

This was never standardized and both Blink and Gecko do not support it.

* LayoutTests/fast/css/getMatchedCSSProperties-rule-mutation-expected.txt: Removed.
* LayoutTests/fast/css/getMatchedCSSProperties-rule-mutation.html: Removed.
* LayoutTests/fast/css/getMatchedCSSRules-crash-expected.txt: Removed.
* LayoutTests/fast/css/getMatchedCSSRules-crash.html: Removed.
* LayoutTests/fast/css/modify-ua-rules-from-javascript-expected.txt: Removed.
* LayoutTests/fast/css/modify-ua-rules-from-javascript.html: Removed.
* LayoutTests/fast/dom/StyleSheet/detached-parent-rule-without-wrapper-expected.txt:
* LayoutTests/fast/dom/StyleSheet/detached-parent-rule-without-wrapper.html:
* LayoutTests/fast/dom/Window/get-set-properties-expected.txt:
* LayoutTests/fast/dom/Window/get-set-properties.html:
* LayoutTests/fast/dom/Window/getMatchedCSSRules-null-crash-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-null-crash.html: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-invalid-pseudo-elements-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-invalid-pseudo-elements.html: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements-complex-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements-complex.html: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements-expected.txt: Removed.
* LayoutTests/fast/dom/Window/getMatchedCSSRules-with-pseudo-elements.html: Removed.
* LayoutTests/fast/dom/Window/resources/WindowProperties.js:
* LayoutTests/fast/dom/Window/resources/window-properties.js:
* LayoutTests/fast/dom/Window/window-function-frame-getter-precedence-expected.txt:
* LayoutTests/fast/dom/Window/window-function-name-getter-precedence-expected.txt:
* LayoutTests/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/fast/dom/Window/window-property-shadowing-expected.txt:
* LayoutTests/fast/dom/Window/window-property-shadowing.html:
* LayoutTests/fast/dom/collection-iterators.html:
* LayoutTests/fast/dom/defaultView-expected.txt:
* LayoutTests/fast/dom/defaultView.html:
* LayoutTests/fast/dom/resources/wrapper-identity-base.js:
* LayoutTests/fast/dom/wrapper-identity-expected.txt:
* LayoutTests/fast/inspector-support/matchedrules.html: Removed.
* LayoutTests/fast/selectors/ignore-pseudo-element-inside-non-rightmost-fragments-expected.txt: Removed.
* LayoutTests/fast/selectors/ignore-pseudo-element-inside-non-rightmost-fragments.html: Removed.
* LayoutTests/http/tests/security/cross-frame-access-call-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-call.html:
* LayoutTests/http/tests/security/cross-frame-access-get-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-get.html:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor.html:
* LayoutTests/http/tests/security/cross-frame-access-put-expected.txt:
* LayoutTests/http/tests/security/cross-frame-access-put.html:
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules.html: Removed.
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules2-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-getMatchedCSSRules2.html: Removed.
* LayoutTests/http/tests/security/resources/cross-frame-iframe-for-put-test.html:
* LayoutTests/http/tests/security/resources/cross-origin-getMatchedCSSRules-frame.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/compat/historical-expected.txt:
* LayoutTests/js/dom/global-function-resolve-expected.txt:
* LayoutTests/js/dom/script-tests/global-function-resolve.js:
* LayoutTests/platform/gtk/fast/dom/Window/window-lookup-precedence-expected.txt:
* LayoutTests/platform/mac-wk2/fast/dom/Window/window-lookup-precedence-expected.txt:
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
(WebCore::DOMWrapperWorld::isUser const):
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/page/DOMWindow.idl:

Canonical link: https://commits.webkit.org/252542@main
